src/main/java/games/strategy/engine/ClientFileSystemHelper.java

```diff
@@ -30,66 +29,51 @@ private ClientFileSystemHelper() {}
-    if (classFilePath.contains(jarFileName)) {
-      return getRootFolderRelativeToJar(classFilePath, jarFileName);
-    }
-
-    return getRootFolderRelativeToClassFile(classFilePath);
-  }
-
-  private static String getGameRunnerClassFilePath() {
-    final String runnerClassName = GameRunner.class.getSimpleName() + ".class";
-    final URL url = GameRunner.class.getResource(runnerClassName);
-    final String path = url.getFile();
     try {
-      // Deal with spaces in the file name which would be url encoded
-      return URLDecoder.decode(path, StandardCharsets.UTF_8.name());
-    } catch (final UnsupportedEncodingException e) {
-      throw new AssertionError("platform does not support UTF-8 charset", e);
+      return getFolderContainingFileWithName(getCodeSourceFolder(), GAME_ENGINE_PROPERTIES_FILE);
+    } catch (final IOException e) {
+      throw new IllegalStateException("unable to locate root folder", e);
     }
   }
 
-  private static File getRootFolderRelativeToJar(final String path, final String jarFileName) {
-    final String subString = path.substring(
-        "file:/".length() - (SystemProperties.isWindows() ? 0 : 1),
-        path.indexOf(jarFileName) - 1);
-    final File f = new File(subString).getParentFile();
-    if (!f.exists()) {
-      throw new IllegalStateException("File not found:" + f);
+  private static File getCodeSourceFolder() throws IOException {
+    final @Nullable CodeSource codeSource = ClientFileSystemHelper.class.getProtectionDomain().getCodeSource();
+    if (codeSource == null) {
+      throw new IOException("code source is not available");
     }
-    return f;
-  }
 
-  private static File getRootFolderRelativeToClassFile(final String path) {
-    File f = new File(path);
-
-    // move up one directory for each package
-    final int moveUpCount = GameRunner.class.getName().split("\\.").length + 1;
-    for (int i = 0; i < moveUpCount; i++) {
-      f = f.getParentFile();
+    final File codeSourceLocation;
+    try {
+      codeSourceLocation = new File(codeSource.getLocation().toURI());
+    } catch (final URISyntaxException e) {
+      throw new IOException("code source location URI is malformed", e);
     }
 
-    // keep moving up one directory until we find the game_engine properties file that we expect to be at the root
-    while (!folderContainsGamePropsFile(f)) {
-      f = f.getParentFile();
-    }
+    // code source location is either a jar file (installation) or a folder (dev environment)
+    return codeSourceLocation.isFile() ? codeSourceLocation.getParentFile() : codeSourceLocation;
+  }
 
-    if (!f.exists()) {
-      System.err.println("Could not find root folder, does  not exist:" + f);
-      return new File(SystemProperties.getUserDir());
+  @VisibleForTesting
+  static File getFolderContainingFileWithName(final File startFolder, final String fileName) throws IOException {
+    @Nullable
+    File folder;
```

Comment:
Declaring this variable outside the loop was necessary to avoid the Eclipse formatter introducing a Checkstyle indentation violation.  (Ask if you want to know the gory details. :smile:)